nncf/quantization/algorithms/min_max/algorithm.py

```diff
@@ -362,33 +349,38 @@ def _apply(self,
         model_transformer = ModelTransformerFactory.create(model)
 
         quantization_target_points = self._get_quantization_target_points(model)
-        weight_tensor_names = set()
+        weight_layer_names = set()
+        def filter_func(point):
+            return MinMaxQuantization in point.algorithm_to_tensor_collectors and \
+                   point.target_point.type == quantization_target_point.type
 
         for quantization_target_point, qconfig in quantization_target_points.items():
             target_node_name = quantization_target_point.target_node_name
-            node = nncf_graph.get_node_by_name(target_node_name)
-            if quantization_target_point.type == TargetType.OPERATION_WITH_WEIGHTS:
-                weight_tensor_name, weight_tensor = self._backend_entity.get_weight_tensor(model,
-                                                                                           quantization_target_point)
-                # If the nodes share one weight tensor, we should have only one quantizer on that
-                if weight_tensor_name in weight_tensor_names:
-                    continue
-                weight_tensor_names.add(weight_tensor_name)
-                command = self._backend_entity.create_weight_quantizer_insertion_command(quantization_target_point,
-                                                                                         qconfig, weight_tensor, node)
-                transformation_commands.append(command)
-            elif quantization_target_point.type in [TargetType.PRE_LAYER_OPERATION, TargetType.POST_LAYER_OPERATION]:
-                def filter_func(point):
-                    return MinMaxQuantization in point.algorithm_to_tensor_collectors and \
-                           point.target_point.type == quantization_target_point.type
-
-                for tensor_collector in statistic_points.get_algo_statistics_for_node(target_node_name, filter_func,
-                                                                                      MinMaxQuantization):
+            for tensor_collector in statistic_points.get_algo_statistics_for_node(
+                    target_node_name,
+                    filter_func,
+                    MinMaxQuantization):
+                if quantization_target_point.is_weight_target_point():
+                    # If the nodes share one weight tensor, we should have only one quantizer on that
+                    layer_name = nncf_graph.get_node_by_name(target_node_name).layer_name
+                    if layer_name in weight_layer_names:
```

Comment:
Is it possible to have and use the NNCFNode attribute `is_shared`?

Reply:
`is_shared` attribute is not enough: as we need to add one FQ for each unique weight we need to distinguish layers with shared weights. Layer name here represents unique weight id. Please see ticket 102810